### PR TITLE
Add channel avatar to sidebar

### LINKED ISSUE #928: Navigation: Sections for "Channels you moderate" & "Channels"
As a channel moderator, I'd like an easy way to see which channels I am a moderator on and which channels I'm subscribed to. 

In the navigation, add a section for "Channels you own" (channels for which the current user is a moderator) and a section for "Channels" (the channels to which the current user is subscribed). For "Channels you own" there should also be a gear icon for each channel which links to the channel settings interface. 

See https://invis.io/9EN908ZQPAX#/311617712_Channel_Settings_Entrance

### acceptance criteria:
- [ ] move 'Submit a new post' button below the home tab

### out of scope
- sorting channels in the navigation

### LINKED ISSUE #1061: Navigation: channel avatars
Related to #927 

As a user, I'd like to have visual branding and reinforcement for channels in the left nav, to make them easier to recognize. 

#927 allows channel moderators to upload avatars. This issue displays the avatars in the left navigation. 

### questions
- does this require a new or modified channel API?

### related issues
- #989 update the channel listing template to use the banner and avatar

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #928 
Fixes #1061 

#### What's this PR do?
Adds channel avatars to sidebar and does some refactoring to make it look more like the mockup

#### How should this be manually tested?
View the channel sidebar and verify that the channels are grouped according to moderator status. Channels you're moderator of should have gear buttons which open the corresponding settings page for that channel

### CODE REVIEW

**Comment by aliceriot** [comment]
Is the gear icon on the channels I own the appearance we want? In the mockups it looks like there is an 'edit' button instead:

edit_climatex

but on the linked issue @pdpinch suggested the gear icon - which is the signed-off design? personally I think the gear icons are a little visually jarring:

gear_icons

I think while we're here we should probably remove the top and bottom borders, since they are removed in the mockups

**Comment by aliceriot** [comment]
I think the copy should also be 'Channels you own' not 'Channels you moderate'

**Comment by noisecapella** [comment]
@pdpinch said it should be 'Channels you moderate' here: https://github.com/mitodl/open-discussions/issues/928#issuecomment-412968813

**Comment by noisecapella** [comment]
@pdpinch What should I do regarding the gear icon?

**Comment by pdpinch** [comment]
@aliceriot thanks for checking the details. 

- We definitely want to standardize on "moderators". I've asked Lucas to update the designs to reflect that, but he hasn't gotten to it yet. 

- The gear is gone and the design is correct: just an `edit` button

**Comment by noisecapella** [comment]
Any other comments @aliceriot?

**Comment by aliceriot** [comment]
yeah, sorry I managed to mess up my reddit somehow so I was fixing that, will take a look soon

File: static/js/components/SubscriptionsList.js [hunk 1]
```diff
@@ -14,19 +16,54 @@ const channelClassName = (channelName, currentChannel) =>
   currentChannel === channelName ? "location current-location" : "location"
 
 export default class SubscriptionsList extends React.Component<Props> {
+  makeChannelLink = R.curry((channel: Channel) => {
```
Comment (aliceriot) [inline static/js/components/SubscriptionsList.js]: does this need to be a curried function? it's only got one argument

Comment (noisecapella) [reply static/js/components/SubscriptionsList.js]: No, at some point I wanted to pass in an extra parameter for whether the channel is moderated by the user or not, but then I remembered I could use `channel.user_is_moderator` so it was unnecessary. I'll remove it

**Pull request merged**